diff --git a/src/firejail/pulseaudio.c b/src/firejail/pulseaudio.c
index 84cbb19..61c34df 100644
--- a/src/firejail/pulseaudio.c
+++ b/src/firejail/pulseaudio.c
@@ -113,6 +113,16 @@ void pulseaudio_init(void) {
 	if (set_perms(RUN_PULSE_DIR, getuid(), getgid(), 0700))
 		errExit("set_perms");
 
+        // also create client.conf.d for dynamically change support
+        // can use firejail --put to dynamically change pulse server.
+        char *confdir = NULL;
+	if (asprintf(&confdir, "%s/client.conf.d", RUN_PULSE_DIR) == -1)
+		errExit("asprintf");
+	if (mkdir(confdir, 0700) == -1)
+		errExit("mkdir client.conf.d");
+	if (set_perms(confdir, getuid(), getgid(), 0700))
+		errExit("set_perms client.conf.d");
+
 	// create ~/.config/pulse directory if not present
 	char *homeusercfg = NULL;
 	if (asprintf(&homeusercfg, "%s/.config", cfg.homedir) == -1)
